presto-main/src/test/java/io/prestosql/operator/scalar/TestMathFunctions.java

```diff
@@ -688,11 +688,19 @@ public void testRandom()
         functionAssertions.tryEvaluateWithAll("rand()", DOUBLE, TEST_SESSION);
         functionAssertions.tryEvaluateWithAll("random()", DOUBLE, TEST_SESSION);
         functionAssertions.tryEvaluateWithAll("rand(1000)", INTEGER, TEST_SESSION);
+        functionAssertions.tryEvaluateWithAll("random(TINYINT'3', TINYINT'5')", TINYINT, TEST_SESSION);
+        functionAssertions.tryEvaluateWithAll("random(SMALLINT'20000', SMALLINT'30000')", SMALLINT, TEST_SESSION);
+        functionAssertions.tryEvaluateWithAll("random(1000, 2000)", INTEGER, TEST_SESSION);
+        functionAssertions.tryEvaluateWithAll("random(-10, -5", INTEGER, TEST_SESSION);
+        functionAssertions.tryEvaluateWithAll("random(-10, 10", INTEGER, TEST_SESSION);
         functionAssertions.tryEvaluateWithAll("random(2000)", INTEGER, TEST_SESSION);
         functionAssertions.tryEvaluateWithAll("random(3000000000)", BIGINT, TEST_SESSION);
+        functionAssertions.tryEvaluateWithAll("random(3000000000, 5000000000)", BIGINT, TEST_SESSION);
 
         assertInvalidFunction("rand(-1)", "bound must be positive");
         assertInvalidFunction("rand(-3000000000)", "bound must be positive");
+        assertInvalidFunction("random(1000, 500)", "start value must be less than stop value");
+        assertInvalidFunction("random(500, 500)", "start value must be less than stop value");
```

Comment:
These invalid cases cover only INTEGER. Could you add cases for TINYINT, SMALLINT and BIGINT?